### PR TITLE
Remove config.warn, Node.warn and pytest_logwarning

### LINKED ISSUE #3078: Remove config.warn, Node.warn and pytest_logwarning
Follow up of #2452

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #3078

### CODE REVIEW

File: src/_pytest/hookspec.py [hunk 1]
```diff
@@ -496,27 +496,6 @@ def pytest_terminal_summary(terminalreporter, exitstatus):
     """
 
 
-@hookspec(historic=True)
-def pytest_logwarning(message, code, nodeid, fslocation):
```
Comment (RonnyPfannschmidt) [inline src/_pytest/hookspec.py]: we should keep this hook in and issue a warning about it not being called anymore, sheduling that removal for later

Comment (nicoddemus) [reply src/_pytest/hookspec.py]: Hmm when do we issue the warning?

Comment (RonnyPfannschmidt) [reply src/_pytest/hookspec.py]: we warn for implementations of the hook using warn_on_impl of pluggy

Comment (nicoddemus) [reply src/_pytest/hookspec.py]: Oh right. Shame we forgot to do that earlier...

Comment (nicoddemus) [reply src/_pytest/hookspec.py]: Done!

**Comment by nicoddemus** [comment]
Fixed now (hopefully)

File: src/_pytest/deprecated.py [hunk 2]
```diff
@@ -110,3 +113,7 @@
     "pytest/tmpdir_factory.ensuretemp is deprecated, \n"
     "please use the tmp_path fixture or tmp_path_factory.mktemp"
 )
+
+PYTEST_LOGWARNING = PytestDeprecationWarning(
+    "pytest_logwarning is deprecated and will be removed soon, use pytest_warning_captured instead"
```
Comment (RonnyPfannschmidt) [inline src/_pytest/deprecated.py]: this message should include the critical information that the hook is no longer being called

Comment (nicoddemus) [reply src/_pytest/deprecated.py]: True

Comment (nicoddemus) [reply src/_pytest/deprecated.py]: Fixed! 👍

**Comment by nicoddemus** [comment]
Ready for additional review/merge

**Comment by nicoddemus** [comment]
Merging as the failure is unrelated.

**Pull request merged**